server/services/BarChartTableService/index.js

```diff
@@ -33,36 +30,30 @@ class BarChartTableService {
 
   csvTableData = () => {
     return {
-      tableColumns: this.websiteColumns.map(({ name }) => name),
-      tableRows: this.websiteRows.map((siteData) =>
-        siteData.map(({ data }) => data)
-      ),
+      tableColumns: [SITE, ...this.websiteColumns.map(({ name }) => name)],
+      tableRows: this.websiteRows.map((siteData) => Object.values(siteData)),
     }
   }
 
   _graphTableColumns = () => [
-    { name: SITE, color: 'gray' },
     ...this.labels
       .filter((column) => column.name !== N_A)
-      .concat({ name: TOTAL_LABEL, color: 'gray' }),
+      .concat({ name: TOTAL_LABEL }),
   ]
 
-  _graphTableRowData = (sortedGraphTableData, tableHeaders) => {
+  _graphTableRowData = (sortedGraphTableData) => {
     return sortedGraphTableData.map((siteData) => {
-      return tableHeaders.map(({ name: columnHeader, color }) => {
-        const {
-          name,
-          counts: { [columnHeader]: siteCount },
-          targets: { [columnHeader]: siteTarget },
-        } = siteData
-
-        return columnHeader === SITE
-          ? { data: name, color }
-          : {
-              data: this._formatGraphTableCellData(siteTarget, siteCount),
-              color,
-            }
-      })
+      return {
+        site: siteData.name,
+        ...Object.keys(siteData.counts).reduce((varData, nextVar) => {
+          varData[nextVar] = this._formatGraphTableCellData(
+            siteData?.targets?.[nextVar],
```

Comment:
why are we using the safe navigator for siteData here but not on the line below for counts?

Reply:
Testing different iterations and for targets it came up as being undefined and it kept erroring out.

Reply:
I get that for targets but you shouldn't need it for `siteData`

```suggestion
            siteData.targets?.[nextVar],
```

Reply:
[e61d2c6](https://github.com/AMP-SCZ/dpdash/pull/695/commits/e61d2c603020119c39c8112e6167441b83a0a61a)